content/doc/book/using/agent-master-access-control.adoc

```diff
@@ -0,0 +1,475 @@
+
+[[AgentToMasterAccessControl-VettingSlaveToMasterCallable/SlaveToMasterFileCallable]]
+=== Vetting `+SlaveToMasterCallable+`/`+SlaveToMasterFileCallable+`
+
+When marking `+Callable+` for agent → master, care has to be taken to
+ensure that the implementation is not exploitable by malicious agents.
+
+* A malicious agent controls the Java serialization payload, so when
+your `+Callable+` gets deserialized on the master, all the serialized
+fields are controlled by the agent.
+* An agent does not control class definitions on the master, so you can
+trust all the classes and methods to behave as it is written. It is not
+possible for a malicious agent to change the code executed on the
+master.
+
+For example, the following `+SlaveToMasterCallable+` is exploitable.
+Callable itself is not public, but a malicious agent can send in
+arbitrary `+path+`, so it can be used to read any file on the master:
+
+[source,groovy]
+----
+// UNSAFE
+class SomeCodeThatRunsOnAgent {
+    void readBackSomeFileFromMaster() {
+        final String path = "...";
+        channel.call(new SlaveToMasterCallable<String,IOException>() {
+            public String call() {
+                return FileUtils.readFileToString(new File(path));
+            }
+        });
+    }
+}
+----
+
+`+Callable+` that delegates execution to a deserialized object is
+dangerous and needs to be carefully examined, because a malicious agent
+can designate unintended `+Runnable+` object:
+
+[source,groovy]
+----
+// UNSAFE
+class MyCallable extends SlaveToMasterCallable<Void> {
+    Runnable r;
+    public Void call() {
+        r.run();
+        return null;
+    }
+}
+----
+
+To avoid this hassle entirely, consider rewriting your code not to call
+back to a master from an agent. Instead, when a master first sends a
+command to an agent, you can carry all the data you'll need with you.
+This may not be always possible or practical, but it's a lot easier to
+secure.
+
+[[AgentToMasterAccessControl-FixingpluginswithoutrequiringnewerJenkins]]
+=== Fixing plugins without requiring newer Jenkins
+
+Classifying `+Callable+`/`+FileCallable+` requires new classes added to
+Jenkins 1.587/1.580.1. This poses a challenge if you want to retain
+backward compatibility with earlier versions of Jenkins.
+
+To solve this problem, we've developed
+https://github.com/jenkinsci/SECURITY-144-compat[SECURITY-144-compat]
+module. This module let you classify `+Callable+`, while still
+functioning correctly on earlier versions of Jenkins. See the
+https://github.com/jenkinsci/SECURITY-144-compat/blob/master/README.md[documentation
+of `+SECURITY-144-compat+`] for details.
+
+Note
+
+As of version 1.1, this library is deprecated, as its use caused some
+unresolved problems
+(https://issues.jenkins-ci.org/browse/JENKINS-25625[JENKINS-25625]). +
+Anyway 1.580.1 is now a fairly conservative choice of baseline: you will
+not exclude so many users by requiring it for new plugin releases.
```

Comment:
Remove paragraphs that describe ancient history and a workaround for ancient history.
```suggestion
```